### ISSUE TITLE
Tests SIGSEGV

### ISSUE THREAD

**Issue by rupertnash** [otbrown/LAMBReX#3]
```
$ ctest
Test project /Users/rupe/working/hilemms/lambrex-test/build/lambrex
    Start 1: Single Value Initialisation
1/2 Test #1: Single Value Initialisation ......***Failed    0.33 sec
    Start 2: Single Value Initialisation
2/2 Test #2: Single Value Initialisation ......***Failed    0.13 sec

0% tests passed, 2 tests failed out of 2

Total Test time (real) =   0.48 sec

The following tests FAILED:
	  1 - Single Value Initialisation (Failed)
	  2 - Single Value Initialisation (Failed)
Errors while running CTest
```

And trying the catch exe directly:
```
$ bin/c2InitTests
MPI initialized with 1 MPI processes
AMReX (19.02-2-g7d428cb32120) initialized
AMReX (19.02-2-g7d428cb32120) finalized
*** The MPI_Comm_dup() function was called after MPI_FINALIZE was invoked.
*** This is disallowed by the MPI standard.
*** Your MPI job will now abort.
[mbp13-rnash2-1.epcc.ed.ac.uk:17459] Local abort after MPI_FINALIZE started completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```

We can try to debug next week. Cheers!

**Comment by otbrown** [comment]
Hi Rupert,

I'm not entirely surprised -- I've not tested with MPI turned on at all yet. This *should* work if you build AMReX with ENABLE_MPI=OFF, but it'll be useful to work out why this occurs regardless. Might provide some useful insight ahead of actually turning on MPI!

**Comment by otbrown** [comment]
Haven't checked yet, but this is almost certainly a consequence of the way Catch runs Test Cases. It reruns everything in `TEST_CASE` scope but each `SECTION` only once, currently `amrex::Initialize` and `amrex::Finalize` are called in `TEST_CASE` scope. Repeat calls to both are fine in serial but follow MPI initialization rules when MPI is included in AMReX.

Possible solutions:
- Move `Initialize` and `Finalize` to their own sections - as long as Catch always respects the order of sections (and I'm sure it does) this should be fine.
- Change to more traditional class based test fixtures with setup and teardown, which Catch also supports. 

I'm inclined to solution 1 for its simplicity, will implement and test ASAP.

**Comment by otbrown** [comment]
Rupert pointed out to me that creating Initialize and Finalize sections required one `TEST_CASE` per executable. Luckily Catch provides a way of doing test run set-up and teardown through event listeners. Amrex (and MPI) are now initialised and finalised at the beginning and end of a test run. Will push and close.

Closed by 6599d81 and f82a0e0.
